### ISSUE TITLE
SEOMatic preview and metatags stopped working after update plugin and cms

### ISSUE THREAD

**Issue by Robsondj** [nystudio107/craft-seomatic#1257]
### Describe the bug
After updating the craft cms, SEOMatic, and others plugins, the SEOMatic preview stopped working as shown in the images below.
The metatags on the site stopped being returned too. Data for metatags are returned by GraphQL API on the site

### To reproduce
1. Update plugins versions in composer.json as shown in the image
2. Run composer update
3. Copy web/index.php bootstrap.php e craft from an updated version to your craft
4. Run `php craft setup/security-key` 
5. Open the craft admin and conclude the process of updating by clicking on Terminate. This will update the database
6. After that, everything works well except the preview and metatags as explained above

### Expected behavior
After updating I expected SEOMatic to continue to work as before the update.

### Screenshots

- The old and new versions of what was updated

<img width="1671" alt="image" src="https://user-images.githubusercontent.com/621503/210138996-8ca0c506-ff82-469f-a48f-e4b7dc9d9ce3.png">
<br>

- Twitter Tab configuration and preview that is not working. The same happens in Google and Facebook Tabs

<img width="2258" alt="image" src="https://user-images.githubusercontent.com/621503/210139084-75ad56a3-0f9e-4dd2-9b4f-f74d50676eba.png">
<br>

- Metatags on the site that were returned with configured information before the updating

<img width="752" alt="image" src="https://user-images.githubusercontent.com/621503/210139189-16bd21e5-ca6c-4ec0-9e3a-885a582bfe56.png">

### Versions
- Plugin version: 4.0.12
- Craft version: 4.2.7

**Comment by khalwat** [comment]
You state that you conclude the process of updating by clicking on **Terminate** but that doesn't sound right -- that sounds like perhaps you stopped a process before it was actually finished?

Other than compatibility with Craft 4, there weren't any major changes in the schema/settings that SEOmatic uses, so I'm thinking something else might have gone awry in your update from Craft 3 to Craft 4.

**Comment by Robsondj** [comment]
@khalwat the Terminate option appeared because I updated cms and plugin's versions via compose, so when I opened the craft admin this option was shown

**Comment by khalwat** [comment]
Something doesn't sound right -- perhaps you didn't run migrations via the CLI after updating?

Can you give me a screenshot of this "terminate" button?

In any event, I'm not sure your issue here is specific to SEOmatic, it seems like something in the Craft update didn't work fully to me.

You're saying everything else works, the only issue you're having is what you outline above re: SEOmatic?

The Twitter handle is set in SEOmatic -> Site Settings -> Social Media

**Comment by Robsondj** [comment]
Sorry @khalwat, about the Terminate button, actually the button is **Finish up**. This option concludes the database update. 
![image](https://user-images.githubusercontent.com/621503/210276745-9108b563-5038-48c4-9ad9-4454689ce4f3.png)
<br>
In fact, the only issue I have after the update is what I said above about SEOmatic
The Twitter configuration keeps as it was before the update.

**Comment by khalwat** [comment]
So to be clear, your two problems are:

* SEOMatic preview stopped working as shown in the images below.
* The metatags on the site stopped being returned too. 

So let's address the second item first.

What you say the meta tags "stop being returned" what exactly does that mean? What is being returned?

**Comment by Robsondj** [comment]
@khalwat I mean, before the update, metatags were returned by GraphQL API to be placed on the site. After the update, the metatags stopped being returned by the GraphQL query with the data I put on SEOmatic.
For instance, as shown above, I put  a description and an image for Twitter on SEOmatic, and this data stopped being returned by GraphQL API after the update

We get all data on the site via GraphQL API because the site is developed with ReactJS

**Comment by khalwat** [comment]
Okay, so what _is_ being returned by the GraphQL query exactly? And what is the query?

**Comment by Robsondj** [comment]
@khalwat **this is the GraphQL query:**
`query GetInsightPageBySlug {
	page: entry(
		site: "global"
		section: "insights"
		slug: "insights-default-seomatic"
	) {
		... on insights_insights_Entry {
			id
			postDate
			title
			topic: singleTopic {
				title
				slug
				featuredImage {
					title
					src: url
					extension
					type: mimeType
					kind
					focalPoint
					hasFocalPoint
				}
			}
			tags {
				title
				slug
			}
			authors {
				type: leaderType {
					slug
				}
				name: title
				slug
				uri
			}
			excerpt
			description
			language: dropdownLanguage
			featuredImage {
				title
				src: url
				extension
				type: mimeType
				kind
				focalPoint
				hasFocalPoint
			}
			featuredImageCaption
			seomatic {
				metaTitleContainer
				metaTagContainer
			}
			
		}
	}
}`

**And this is the result that should have a twitter description and image on seomatic session**
`{
	"data": {
		"page": {
			"id": "90326",
			"postDate": "2022-06-02T06:08:00-07:00",
			"title": "Insights default SEOmatic",
			"topic": [
				{
					"title": "topico titulo grandissimo",
					"slug": "travel",
					"featuredImage": [
						{
							"title": "Pablo Picasso",
							"src": "https://web-stage.bcwlabs.com/assets/Pablo-Picasso.jpg",
							"extension": "jpg",
							"type": "image/jpeg",
							"kind": "image",
							"focalPoint": [
								0.5,
								0.5
							],
							"hasFocalPoint": false
						}
					]
				}
			],
			"tags": [
				{
					"title": "TagNew",
					"slug": "tagnew"
				}
			],
			"authors": [
				{
					"type": [
						{
							"slug": "bcw-leader"
						}
					],
					"name": "New BCW Leader",
					"slug": "new-bcw-leader",
					"uri": "leaders/global/new-bcw-leader"
				}
			],
			"excerpt": "This is the excerpt 1",
			"description": "This is the content description",
			"language": "english",
			"featuredImage": [
				{
					"title": "Catmay16",
					"src": "https://web-stage.bcwlabs.com/assets/catmay16.jpg",
					"extension": "jpg",
					"type": "image/jpeg",
					"kind": "image",
					"focalPoint": [
						0.5,
						0.5
					],
					"hasFocalPoint": false
				}
			],
			"featuredImageCaption": "this is the image caption",
			"seomatic": {
				"metaTitleContainer": "<title></title>",
				"metaTagContainer": "<meta name=\"robots\" content=\"none\"><meta content=\"en_US\" property=\"og:locale\"><meta content=\"https://www.instagram.com/bcwglobal\" property=\"og:see_also\"><meta content=\"https://www.youtube.com/channel/UCyu9EFSlm8icYlvFszGlzRQ\" property=\"og:see_also\"><meta content=\"https://www.linkedin.com/company/28993430/\" property=\"og:see_also\"><meta content=\"https://en.wikipedia.org/wiki/Burson_Cohn_%26_Wolfe\" property=\"og:see_also\"><meta content=\"https://www.facebook.com/BCWGlobal\" property=\"og:see_also\"><meta content=\"https://twitter.com/BCWGlobal\" property=\"og:see_also\"><meta name=\"twitter:site\" content=\"@\"><meta name=\"twitter:creator\" content=\"@\">"
			}
		}
	}
}`

**Comment by khalwat** [comment]
hmmm okay so some meta tags are being returned, but not filled in the way you're expecting things to be.

I've seen some migrations from Craft 3 to Craft 4 that have the Sites not properly migrated; can you go to:

Settings -> Sites

And check each site to make sure that the site's Status is **Enabled**, and the **This site has its own base URL** checkbox is checked?

<img width="609" alt="Screenshot 2023-01-03 at 11 40 44 AM" src="https://user-images.githubusercontent.com/7570798/210401311-c1313a9c-64cc-41f3-8037-a2cc5606b4ca.png">

Also if you verify that, then go to:

SEOmatic -> Content SEO

And post a screenshot of what you see there? Are all of the Sections you'd expect to see listed there?

**Comment by Robsondj** [comment]
@khalwat The **Status** is enabled and **This site has its own base URL** is checked

And the Content SEO is filled as expected
<img width="2537" alt="image" src="https://user-images.githubusercontent.com/621503/210431607-986935bf-9bac-4b4c-8894-1c5bf830a249.png">

**Comment by khalwat** [comment]
So I think what we should do at this point is get on a video conference so I can have a look at things, if you're willing.

Do you have a local development environment in which this site is running, and you can reproduce the issue?

If so, book a time here:

https://savvycal.com/nystudio107/chat

**Comment by Robsondj** [comment]
@khalwat we can now see all PHP errors on the stage server
<img width="831" alt="image" src="https://user-images.githubusercontent.com/621503/211081899-f9788ad8-99ec-4bfd-b725-7b583d28c5a3.png">

Can I schedule another meeting with you or is there something more you want me to do?

**Comment by khalwat** [comment]
Great, what was wrong?

Also yes, feel free to use the link above to reschedule.

**Comment by Robsondj** [comment]
We needed to enable display_errors inside the php.ini

Ok, I'll schedule another meeting!
Thank you!

**Comment by khalwat** [comment]
...and is your CP now showing the black & yellow stripped border at the bottom to indicate that `devMode` is indeed enabled?

https://craftcms.com/knowledge-base/what-dev-mode-does

**Comment by Robsondj** [comment]
note yet, but I'm going to investigate it too

**Comment by khalwat** [comment]
Yeah let's get that addressed too, before we meet up again

**Comment by Robsondj** [comment]
@khalwat now we can see the yellow striped border at the bottom. To solve this problem I needed to put `CRAFT_DEV_MODE=true` on .env file
We can also debug on .twig files with `{% dd variable %}`
<img width="222" alt="image" src="https://user-images.githubusercontent.com/621503/211325900-1c6e3726-e4a0-4643-b539-6ca9eec64b05.png">

**Comment by khalwat** [comment]
I'm a little concerned that the changes we were making in the general.php were not taking, but at least you have this working

I am running about 10 to 15 minutes late to the meeting today

**Comment by khalwat** [comment]
Addressed in: https://github.com/nystudio107/craft-seomatic/commit/b8d859afb63dca305c2c26f0e29e1c25f5da1234 & https://github.com/nystudio107/craft-seomatic/commit/8f8a40e931b57673114dd489061cb583c46dbf90

Released as:

Craft CMS 3: Version 3.4.47 -> https://github.com/nystudio107/craft-seomatic/releases/tag/3.4.47

Craft CMS 4: Version 4.0.18 -> https://github.com/nystudio107/craft-seomatic/releases/tag/4.0.18

**Comment by khalwat** [comment]
I was able to reproduce your issue locally with the composer.json that you sent along… but it has SEOmatic locked to this version:

"nystudio107/craft-seomatic": "4.0.18”,

This should really be set to the following, which will pull down the _latest_ SEOmatic:

"nystudio107/craft-seomatic": “^4.0.20”,

(Note the ^ in the server there)

In any event, after a bunch of debugging, I found out the problem:

You have defined a global set named ’SEOmatic’ in your Settings -> Globals -> SEOmatic. This overwrites the injected seomatic variable, causing it to fail to render. Renaming it to SEOmatic Settings (and the Handle to seomaticSettings) fixes the problem.
